src/hades/agent/tasks.py

```diff
@@ -337,10 +348,47 @@ def get_auth_dhcp_leases(
             do_get_all_auth_dhcp_leases(connection, subnet, limit)))
 
 
+@rpc_task()
+def get_unauth_dhcp_leases(
+    subnet: Optional[str] = None,
+    limit: Optional[int] = 100,
+) -> List[Tuple[float, str, str, Optional[str]]]:
+    """Return all unauth leases.
+
+    :param subnet: Limit leases to subnet
+    :param limit: Maximum number of leases
+    :return: A list of (Expires-At, MAC, IP-Address, Hostname,
+        Client-ID)-tuples
+    :raises ArgumentError: if illegal arguments are provided
+    """
+    if subnet is not None:
+        subnet = check_ip_network("subnet", subnet)
+    if limit is not None:
+        limit = check_positive_int("limit", limit)
+    with engine.connect() as connection:
+        leases = do_get_all_unauth_dhcp_leases(connection, subnet, limit)
+    return [
+        (
+            expires_at.timestamp(),
+            str(mac),
+            str(ip),
+            hostname,
+        )
+        for expires_at, mac, ip, hostname in leases
+    ]
```

Comment:
With this change, which introduces an asymmetry between `get_unauth_dhcp_leases` and `get_auth_dhcp_leases`, I'd like to remark on the following, conscious decisions:

1. Using `engine.connect()` directly as a context manager. This most likely just didn't exist in `sqlalchemy<1.3`.
2. Using a list comprehension instead of a `list(starmap(lambda…)))`. IMO it's substantially more readable.
3. Narrowing the scope of the `Connection` context manager. Would have done the same to a `try`: Limits the „safety blanket“ to the relevant code and removes an unnecessary indentation layer, which in general is a good idea to improve code comprehension.